frontend/src/container/GridGraphLayout/Graph/FullView/GraphManager.tsx

```diff
@@ -0,0 +1,242 @@
+import { Button, Input } from 'antd';
+import { CheckboxChangeEvent } from 'antd/es/checkbox';
+import { ColumnType } from 'antd/es/table';
+import { ChartData } from 'chart.js';
+import { ResizeTable } from 'components/ResizeTable';
+import { LOCALSTORAGE } from 'constants/localStorage';
+import { useNotifications } from 'hooks/useNotifications';
+import isEqual from 'lodash-es/isEqual';
+import { memo, useEffect, useState } from 'react';
+
+import CheckBox from './CheckBox';
+import { ColumnsTitle, DataIndexAndKey } from './contants';
+import {
+	FilterTableAndSaveContainer,
+	FilterTableContainer,
+	LabelContainer,
+	SaveContainer,
+} from './styles';
+import { DataSetProps, ExtendedChartDataset, LegendEntryProps } from './types';
+import {
+	getAbbreviatedLabel,
+	getDefaultTableDataSet,
+	saveLegendEntriesToLocalStorage,
+	showAllDataSet,
+} from './utils';
+
+function GraphManager({
+	data,
+	graphVisibilityStateHandler,
+	name,
+}: GraphManagerProps): JSX.Element {
+	const [graphVisibilityState, setGraphVisibilityState] = useState<boolean[]>(
+		Array(data.datasets.length).fill(true),
+	);
+	const [tableDataSet, setTableDataSet] = useState<ExtendedChartDataset[]>(
+		getDefaultTableDataSet(data),
+	);
+	const [legendEntries, setLegendEntries] = useState<LegendEntryProps[]>([]);
+
+	const { notifications } = useNotifications();
+
+	useEffect(() => {
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler([...graphVisibilityState]);
+		}
+	}, [graphVisibilityState, graphVisibilityStateHandler]);
+
+	useEffect(() => {
+		const newGraphVisibilityStates: boolean[] = Array(data.datasets.length).fill(
+			true,
+		);
+		setTableDataSet(getDefaultTableDataSet(data));
+		data.datasets.forEach((d, i) => {
+			const index = legendEntries.findIndex((di) => di.label === d.label);
+			if (index !== -1) {
+				newGraphVisibilityStates[i] = legendEntries[index].show;
+			}
+		});
+		setGraphVisibilityState(newGraphVisibilityStates);
+	}, [data, legendEntries]);
+
+	useEffect(() => {
+		if (localStorage.getItem(LOCALSTORAGE.GRAPH_VISIBILITY_STATES) !== null) {
```

Comment:
we are duplicating the logic here in `frontend/src/container/GridGraphLayout/Graph/WidgetGraphComponent.tsx`

Reply:
Move the logic to utils